### DISCUSSION TITLE
Rename `GetInferenceHelpers` -> `inferRouterTypes`?

### DISCUSSION THREAD

**Discussion by KATT** [trpc/trpc#2825] [Ideas]
**Context:** https://trpc.io/docs/v10/infer-types#additional-dx-helper-type

I think the name `GetInferenceHelpers<AppRouter>` is a bit.. confusing. I want another name. Ideas?

cc @trpc/website-docs @trpc/trpc-core

**Comment by bautistaaa** [comment]
How about InferRouterType (singular)

**Reply by KATT** [reply]
`infer` or `Infer`? 👀

**Comment by juliusmarminge** [comment]
InferTRPCRouter

Would go along with createTRPC* etc

It's not bound to just the AppRouter either

**Reply by KATT** [reply]
`infer` or `Infer`? 👀

**Reply by juliusmarminge** [reply]
maybe do infer actually so that if anyone wants to just reuse the name they can do
```ts
type InferTRPCRouter = inferTRPCRouter<AppRouter>
```

and have their types in uppercase like (probably) all their other types 🤔

**Reply by mmkal** [reply]
What about no prefix, to match builtins like `ReturnType` rather than `inferReturnType` or `InferReturnType`?

```ts
export type Router = RouterTypes<typeof router>
```

**Reply by sachinraja** [reply]
I like `RouterTypes` but if we're doing that, we should also remove the `infer` prefix from the other types we export

**Reply by KATT** [reply]
I like `ReturnTypes` too, following "TS conventions" seems good.

- `infer(.*)<X>` -> `$1Type<X>`

I do like having `RouterTypes` with an `s` at the end though but can't motivate why very well.

**Reply by KATT** [reply]
Here's a nonexhaustive list:



- inferQueryOutput
- inferQueryInput
- inferMutationOutput
- inferMutationInput
- inferErrorShape
- inferInfiniteQueryNames
- inferProcedures
- [..]

**Reply by KATT** [reply]
Actually, for the sake of nice autocompletion, I think we should either smurf these or keep with the `infer`-prefix.

Imagine

```ts
export type Router = RouterTypes<typeof router>
```

Whenever I write `Rou..` I will get autocompletion on both - but since `RouterTypes` is something we only want to use exactly once, I think it should either be:

```ts
export type Router = TRPCRouterTypes<typeof router>;
// or
export type Router = inferRouterTypes<typeof router>;
```

**Reply by KATT** [reply]
Btw, related: #3041
